src/dl_typelib_read_txt.cpp

```diff
@@ -243,12 +250,15 @@ static void dl_load_txt_build_default_data( dl_ctx_t ctx, dl_txt_read_ctx* read_
 	uint8_t* pack_buffer = (uint8_t*)dl_alloc( &ctx->alloc, prod_bytes );
 
 	bool use_fast_ptr_patch = false;
-	err = dl_txt_pack_internal( ctx, def_buffer, pack_buffer, prod_bytes, 0x0, use_fast_ptr_patch );
+	err = dl_txt_pack_internal( ctx, def_buffer_ptr, pack_buffer, prod_bytes, 0x0, use_fast_ptr_patch );
 	if( err != DL_ERROR_OK )
 		dl_txt_read_failed( ctx, read_state, DL_ERROR_INVALID_DEFAULT_VALUE, "failed to pack default-value for member \"%s\" with error \"%s\"",
 															dl_internal_member_name( ctx, member ),
 															dl_error_to_string( err ) );
 
+	if( def_buffer_ptr != def_buffer )
+		dl_free( &ctx->alloc, def_buffer_ptr );
```

Comment:
this would leak on error as dl_txt_read_failed() will call longjmp()

Reply:
I wonder if it is worth to add something arena-like to ctx that can be used to push/pop temp-memory?

Reply:
The convenient way would be to wrap the allocator and capture all allocations made in the longjmp scope so they can be freed.

Another thing this shows is the lack of coverage in the testing. Now that I know about this issue I see more such leaks in the code which are there before this PR.